### PR TITLE
Add example for debugging recursion.

### PR DESCRIPTION
**Pull request by gz**
Also fix yarn build and generally improve recursion section.

### CODE REVIEW

File: docs/sql/string.md [hunk 1]
```diff
@@ -198,8 +198,8 @@ addition to the normal way of `''`.
         </ul>
     </td>
     <td>
-        <code>SPLIT_PART('a|b|c|', '|', 2)</code> => <code>b</code><br>
-        <code>SPLIT_PART('a|b|c|', '|', -2)</code> => <code>c</code><br>
+        <code>SPLIT_PART('a|b|c|', '|', 2)</code> => <code>b</code><br></br>
```
[the remark this answers is not in the document]
Comment (gz) [reply docs/sql/string.md:201]: it does, that's why I added them to fix the `yarn build` not sure why it insists on it

File: docs/sql/recursion.mdx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+## Key Concepts of Recursive Views
+
+### Defining Recursive Views with Forward Declarations
+
+In Feldera SQL, recursive queries require [forward declarations](https://en.wikipedia.org/wiki/Forward_declaration)
+when a view references itself or another view before their definition.
+A forward declaration specifies the view's name, column names, and column types.
+
+```sql
```
Comment (mihaibudiu) [inline docs/sql/recursion.mdx:31]: The text explained the purpose of this view.
Not explaining it makes it harder to read.

File: docs/sql/recursion.mdx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+CREATE MATERIALIZED VIEW CLOSURE AS
+(SELECT * FROM EDGES)
+UNION
+(SELECT * FROM STEP);
+```
+
+In this example we compute a transitive closure:
+
+- `EDGES` represents the edges of a graph.
+- `CLOSURE` computes the transitive closure of the graph by recursively combining edges and paths in `STEP`.
```
Comment (mihaibudiu) [inline docs/sql/recursion.mdx:61]: what is "combining"?

File: docs/sql/recursion.mdx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+For example, if `STEP` were not a local view, it would also require a forward declaration.
+
+The type inferred by the compiler for the view based on the query must
+match exactly the type of the declared view, including the nullability
+of the columns, and the column names.
+
+## Semantics
+
+- Mutually recursive views are computed iteratively until they reach a stable state (i.e., no further changes occur).
```
Comment (mihaibudiu) [inline docs/sql/recursion.mdx:76]: they start as empty.

File: docs/sql/recursion.mdx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+Feldera supports a wide range of operators in recursive queries, including:
+
+- `SELECT`, `WHERE`, `GROUP BY`, `HAVING`
+- All types of `JOINs`
+- Aggregations, `DISTINCT`, `UNION`, `INTERSECT`, `EXCEPT`
+- `UNNEST`, `VALUES`, `PIVOT`
+- Table functions like `HOP` and `TUMBLE`
+
+It is also possible to create a recursive *materialized* view, which memoizes the result of
```
Comment (mihaibudiu) [inline docs/sql/recursion.mdx:240]: it's not really memoization.
It is materialization, as defined elsewhere.

File: docs/sql/recursion.mdx (hunk 2 above)
Comment (gz) [reply docs/sql/recursion.mdx:31]: Are you referring to the text that talked about forward declaration in C?

Comment (mihaibudiu) [reply docs/sql/recursion.mdx:31]: Edges in a graph

**Comment by mihaibudiu** [comment]
so you squash the commits, so it's hard to see what you changed, have to read it all over again

**Review by mihaibudiu — APPROVED** [review]

**Comment by gz** [comment]
>so you squash the commits, so it's hard to see what you changed, have to read it all over again

sorry about that

**Pull request merged**